### PR TITLE
Add command line flags for file based stats storage.

### PR DESCRIPTION
**Pull request by crockpotveggies**
## What changes were proposed in this pull request?

Add custom flags for setting StatsFileStorage and custom path if necessary.

## How was this patch tested?

Since this is UI-related, will need to rebuild JAR and deploy on a cluster to actually test effectiveness.

### CODE REVIEW

File: deeplearning4j-ui-parent/deeplearning4j-play/src/main/java/org/deeplearning4j/ui/play/PlayUIServer.java [hunk 1]
```diff
@@ -79,13 +81,19 @@
...
 
     private Thread uiEventRoutingThread;
-    @Parameter(names = {"-r", "-enableRemote"}, description = "Whether to enable remote or not", arity = 1)
-    private boolean enableRemote;
 
+    @Parameter(names = {"-r", "--enableRemote"}, description = "Whether to enable remote or not", arity = 1)
+    private boolean enableRemote;
 
-    @Parameter(names = {"--uiPort"}, description = "Whether to enable remote or not", arity = 1)
+    @Parameter(names = {"-p", "--uiPort"}, description = "Whether to enable remote or not", arity = 1)
```
Comment (AlexDBlack) [inline deeplearning4j-ui-parent/deeplearning4j-play/src/main/java/org/deeplearning4j/ui/play/PlayUIServer.java]: Was already wrong - but let's fix this description?

File: deeplearning4j-ui-parent/deeplearning4j-play/src/main/java/org/deeplearning4j/ui/play/PlayUIServer.java [hunk 2]
```diff
@@ -79,13 +81,19 @@
...
     private int port = DEFAULT_UI_PORT;
 
+    @Parameter(names = {"-fs", "--enableFileStorage"}, description = "Use file-based stats storage", arity = 1)
+    private boolean enableFileStorage;
+
+    @Parameter(names = {"--customStatsFile"}, description = "Path to create custom stats file", arity = 1)
```
Comment (AlexDBlack) [inline deeplearning4j-ui-parent/deeplearning4j-play/src/main/java/org/deeplearning4j/ui/play/PlayUIServer.java]: arity=1 I think is only valid for boolean args. Also improve description - " for remote listener support. If set, also enables remote listener."

File: deeplearning4j-ui-parent/deeplearning4j-play/src/main/java/org/deeplearning4j/ui/play/PlayUIServer.java [hunk 3]
```diff
@@ -222,7 +230,20 @@ public void runMain(String[] args) {
...
+                File tempStatsFile = File.createTempFile("dl4j", "UIstats");
+                tempStatsFile.deleteOnExit();
+                enableRemoteListener(new FileStatsStorage(tempStatsFile), true);
+            } catch(Exception e) {
+                System.out.println("Failed to create temporary file for stats storage");
+                System.exit(0);
+            }
+        }
+        else if(enableRemote)
             enableRemoteListener();
```
Comment (AlexDBlack) [inline deeplearning4j-ui-parent/deeplearning4j-play/src/main/java/org/deeplearning4j/ui/play/PlayUIServer.java]: On second thought - let's merge enableRemote and enableFileStorage - i.e., default to FileStorage only (temp dir unless path is set). File has better performance and less memory use...

File: deeplearning4j-ui-parent/deeplearning4j-play/src/main/java/org/deeplearning4j/ui/play/PlayUIServer.java (hunk 2 above)
Comment (crockpotveggies) [reply deeplearning4j-ui-parent/deeplearning4j-play/src/main/java/org/deeplearning4j/ui/play/PlayUIServer.java]: incorrect, the javadoc says

> How many parameter values this parameter will consume. For example, an arity of 2 will allow "-pair value1 value2".

**Pull request merged**